fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/DataRegistry.java

```diff
@@ -0,0 +1,93 @@
+package com.github.dedis.popstellar.model.network.method.message.data;
+
+import android.util.Log;
+
+import androidx.core.util.Pair;
+
+import com.github.dedis.popstellar.utility.handler.DataHandler;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+/** A registry of Data classes and handlers */
+public final class DataRegistry {
+
+  /** A mapping of (object, action) -> (class, handler) */
+  private final Map<EntryPair, Pair<Class<? extends Data>, DataHandler<? extends Data>>> mapping;
```

Comment:
I am not a Registry specialist, but usually what I do is to create an inner Entry class that contains everything :
 - Method
- Action
- DataClass
- DataHandler

And then at creation generate multiple mapping given our needs.

Here I would probably create a mapping `EntryPair -> Entry` and probably a mapping `DataClass -> Entry` will also be useful in the near future.

What do you think about it ?